content/blog/2018/06/2018-05-14-simple-pull-request-plugin.adoc

```diff
@@ -0,0 +1,205 @@
+4. Plugin is using Yaml from target branch rigth now. (May be this needs some discussion, example: what if PR contains changes in Jenkinsfile.yaml)
+=== Jenkinsfile.yaml exapmle
+Write now no format has been decided for the yaml file. But for thes demo I am using below yaml Jenkinsfile
+[source,yaml]
+artifactPublishingConfig:
+    host: 192.32.52.12
+    user: user53
+    credentialId: dummyGitRepo
+agent:
+    dockerImage: maven:3.5.3-jdk-8
+    args: -v /tmp:/tmp
+testResultPaths:
+    - target/surefire-reports/*.xml
+findBugs: target/*.xml
+stages:
+    - name: First
+      scripts:
+        -   ./scripts/hello
+    - name: Build
+      scripts:
+        -   ./scripts/build
+    - name: Tests
+      scripts:
+        -   ./scripts/test
+archiveArtifacts:
+    - Jenkinsfile.yaml
+    - scripts/hello.sh
+publishArtifacts:
+    - from: Jenkinsfile.yaml
+      to: ~/archives
+    - from: scripts/hello.sh
+      to: ~/archives
+
+Below pipeline code will be generated using above yaml file.
+
+[source, groovy]
+pipeline {
+  agent {
+    docker {
+      image 'maven:3.5.3-jdk-8'
+      args '-v /tmp:/tmp'
+      alwaysPull false
+      reuseNode false
+    }
+  }
+  stages {
+    stage('First') {
+      steps {
+        script {
+          if (isUnix()) {
+            sh './scripts/hello.sh'
+          } else {
+            bat './scripts/hello.bat'
+          }
+        }
+      }
+    }
+    stage('Build') {
+      steps {
+        script {
+          if (isUnix()) {
+            sh './scripts/build.sh'
+          } else {
+            bat './scripts/build.bat'
+          }
+        }
+      }
+      post {
+        success {
+          archiveArtifacts artifacts: '**/target/*.jar'
+          archiveArtifacts artifacts: 'Jenkinsfile.yaml'
+          archiveArtifacts artifacts: 'scripts/hello.sh'
+        }
+      }
+    }
+    stage('Tests') {
+      steps {
+        script {
+          if (isUnix()) {
+            sh './scripts/test.sh'
+          } else {
+            bat './scripts/test.bat'
+          }
+        }
+      }
+      post {
+        success {
+          junit 'target/surefire-reports/*.xml'
+        }
+        always {
+          findbugs pattern: 'target/*.xml'
+        }
+      }
+    }
+  }
+}
+
+=== Coding Phase 2 plans
+[start=1]
+* Decide a proper YAML format to use for Jenkinsfile.yaml
+* Create Step Configurator for SPRP plugin. https://issues.jenkins-ci.org/browse/JENKINS-51637[JENKINS-51637].
+This will enable users to use Pipeline steps in Jenkinsfile.yaml.
+* Geting rid of manual tab generation in Pipeline SnippetGenerator class.
+* Write tests for the plugin.
+
+
+=== How to reach me
+* Email: gautamabhishek46@gmail.com
+* Gitter room: https://gitter.im/jenkinsci/simple-pull-request-job-plugin
+
+=== References
+
+* https://docs.google.com/document/d/1cuC0AvQG3e4GCjIoCwK3J0tcJVAz1eNDKV8d_zXxlO8/edit[Initial proposal of the project]
+* https://github.com/jenkinsci/simple-pull-request-job-plugin[Project repository]
```

Comment:
Add Gitter chat and project page on jenkins.io?

Reply:
Done